### ISSUE TITLE
Release new minor update with selected fixes

### ISSUE THREAD

**Issue by eschaton** [westes/flex#499]
There hasn’t been a release since 2.6.4 in 2017, but it looks like there have been a few important fixes since then, aside from (and mostly before) a huge bunch of refactoring.

For flex users who want to manage risk, it’d be great to have an update from 2.6.4 that just includes these fixes while mainline continues to get significant work.

**Comment by westes** [comment]
Fair enough. There's work to automate/simplify the release process, which I had meant to get to finalizing, but that hasn't happened yet. That being said, there's nothing stopping anyone from simply aplying patches to the version of flex they started with. 

 I find that people still want the lastest thing if there's any sort of release at all for flex, so publishing a release of flex doesn't reduce the support burden.

It's also worth noting that a lot of the more recent work adds significant new features to flex and not is not solely refactoring.

**Comment by seanm** [comment]
@westes any news on getting a new release? (We've passed a half decade now with no release.)

**Comment by thesamesam** [comment]
The problem is, as a distribution, you have to be able to decide which patches are safe and desirable to backport. For critical toolchain components or where the number of patches is large, or their changes are nontrivial, we generally avoid doing it at all.

I don't know what people want the latest thing, but I can assure you that distributions would be happy enough with a bugfix release.

**Comment by westes** [comment]
@thesamesam definitely a god point; this is on my list to make happen.

**Comment by kberry** [comment]
speaking as one of the maintainers of TeX Live, where we use flex in various ways, may I add my +1(0000) for making a new release of flex. Thanks.

**Comment by seanm** [comment]
>I don't know what people want the latest thing...

For myself, I'd like to see a new flex release because the code that the current version (2.6.4) generates triggers many compiler warnings.  Since 2017, compilers and static analyzers have been improving their warnings, and many of the projects I work with have increased the number of warnings they enable in their CI.  There are several warning flags (that I don't recall now) that generate no warnings in the codebase *except* for files created by flex.

**Comment by gruenich** [comment]
@westes Is there anything we as the community can help you creating a new release?

**Comment by trofi** [comment]
`nixpkgs` has to apply backported patches and pull in `autoconf`, `help2man` and other tools to generate build system and build `flex`. Given that `flex` is needed to build development versions of `gcc` it would be nice to avoid pulling in `autoconf` so early during initial system build bootstrap for very fresh `gcc`.

+1 for an upstream release with fixes applied over the years.
